bob-access/src/extractor.rs

```diff
@@ -14,7 +16,112 @@ impl StubExtractor {
 }
 
 impl<Request> Extractor<Request> for StubExtractor {
-    fn extract(&self, req: &Request) -> Result<Credentials, Error> {
+    fn extract(&self, _req: &Request) -> Result<Credentials, Error> {
         Ok(Credentials::default())
     }
 }
+
+#[derive(Debug, Clone, Default)]
+pub struct BasicExtractor {}
+
+impl<T> Extractor<TRequest<T>> for BasicExtractor {
+    fn extract(&self, req: &TRequest<T>) -> Result<Credentials, Error> {
+        let meta = req.metadata();
+        let username = meta
+            .get("username")
+            .ok_or_else(|| Error::credentials_not_provided("username"))?
+            .to_str()
+            .map_err(|e| Error::unknown(e.to_string()))?;
```

Comment:
Why unknown? Probably should be "parse error" or "conversion error"

Reply:
Replaced with `ConversionError`